projects/graph/graph.py

```diff
@@ -7,49 +7,198 @@ class Graph:
+        Pop first node out of queue
+        If not visited
+        Mark as visited
+        Get adjacent edges and add to list
+        Goto top of loop
+
         """
+        que = Queue()
+        visited = set()
+        que.enqueue(starting_vertex)
+        while que.size() > 0:
+            vertex = que.dequeue()
+            if vertex not in visited:
+                visited.add(vertex)
+                print(vertex)
+                for next_vert in self.vertices[vertex]:
+                    que.enqueue(next_vert)
+
         pass  # TODO
     def dft(self, starting_vertex):
         """
         Print each vertex in depth-first order
         beginning from starting_vertex.
+
+        Make a stack
+        Make visited list
+        Add first node to stack
+        While stack not empty:
+        Pop top item
+        if not visited:
+        Mark as visited
+        Get adjacent and add to stack
+
         """
-        pass  # TODO
+        questack = Stack()
+        visited = set()
+        questack.push(starting_vertex)
+        while questack.size() > 0:
+            vertex = questack.pop()
+            if vertex not in visited:
+                visited.add(vertex)
+                print(vertex)
+                for next_vert in self.vertices[vertex]:
+                    questack.push(next_vert)
+                    #print(f'{next_vert} aaaaAAAA')
+                    #print(visited)
+                    #print(self.vertices)
+                    #print(len(self.vertices))
+                    #print(len(visited))
+                    
+    
     def dft_recursive(self, starting_vertex):
         """
         Print each vertex in depth-first order
         beginning from starting_vertex.
         This should be done using recursion.
         """
-        pass  # TODO
+        if len(self.visited2) == len(self.vertices):
+            return 0
+        if starting_vertex not in self.visited2:
+            self.visited2.add(starting_vertex)
+            if starting_vertex is None:
+                pass
+            else:
+                print(starting_vertex)
+            for next_vert in self.vertices[starting_vertex]:
+                self.dft_recursive(next_vert)
+        else:
+            pass
+            
+        # figure out exit condition (when are we finished)
+        # call itself, simplest case
+        # I want to say all dft_recursive should be doing is checking if it can go deeper, when it can't, mark node as visisted and pass it up.
+        # So you'd start, go deeper, go deeper, go deeper, can't go deeper, mark visited, go back, mark visited, go back, mark visited, done.
+        # Something like that. dft_recursive is essentially go deeper, if not, mark visited and grab the last thing off the call stack.
+        # What's go deeper?
+        # Going deeper is pushing it into visited if it's not visited
+        # exit condition can be when visited and self.vertices have the same length
```

Comment:
can you think of an exit condition that relies only on your visited list and the current vertex?